_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,433 @@
+        oauth2:
+          client:
+            provider:
+              oidc:
+                issuer-uri: https://<your-okta-domain>/oauth2/default
+            registration:
+              oidc:
+                client-id: <client-id>
+                client-secret: <client-secret>
+```
+
+Enable the OIDC authentication in the `jhipster-registry` service by adding the `oauth2` profile in the `kubernetes/registry-k8s/jhipster-registry.yml` file:
+
+```yml
+- name: SPRING_PROFILES_ACTIVE
+  value: prod,k8s,oauth2
+```
+
+## Run locally with Minikube
+
+Install [Minikube](https://minikube.sigs.k8s.io/docs/start/) and [`kubectl`](https://kubernetes.io/docs/tasks/tools/).
+
+For Minkube, you will need at least 2 CPUs. Start MiniKube with your number of CPUs:
+
+```bash
+$ cd kubernetes
+$ minikube --cpus <ncpu> start
+```
+Minikube will log the Kubernetes and Docker versions on start:
+
+```
+Preparing Kubernetes v1.23.3 on Docker 20.10.12 ...
+```
+
+For the `store-mongodb` deployment to work, the property `Service.spec.publishNotReadyAddresses` was required, instead of the annotation `service.alpha.kubernetes.io/tolerate-unready-endpoints`, as the latter was deprecated in [Kubernetes release 1.11](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.11.md#kubernetes-111-release-notes).
+
+Edit `kubernetes/store-k8s/store-mongodb.yml` and add the `publishNotReadyAddresses: true` property to the `spec` key near the bottom of the file:
+
+```yaml
+...
+# Headless service for DNS record
+apiVersion: v1
+kind: Service
+metadata:
+  name: store-mongodb
+  namespace: demo
+spec:
+  type: ClusterIP
+  clusterIP: None
+  publishNotReadyAddresses: true
+  ports:
+    - name: peer
+      port: 27017
+  selector:
+    app: store-mongodb
+```
+
+Then deploy the application to Minikube, in the `kubernetes` directory, run:
+
+```bash
+$ ./kubectl-apply.sh -f
+```
+Now is a good time to install [k9s](https://k9scli.io/topics/install/), a terminal-based UI to interact with Kubernetes clusters. Then run k9s with:
+
+```bash
+$ k9s -n demo
+```
+
+{% img blog/jhipster-digital-ocean/k9s-pods.png alt:"k9s UI" width:"800" %}{: .center-image }
+
+Check out the [commands list](https://k9scli.io/topics/commands/), some useful ones are:
+- `:namespace`: show all available namespaces
+- `:pods`: show all available pods
+You can navigate the pods with `ENTER` and go back with `ESC` keys.
+
+Set up port-forwarding for the JHipster Registry.
+
+```bash
+$ kubectl port-forward svc/jhipster-registry -n demo 8761
+```
+
+Navigate to `http://localhost:8761` and sign in with your Okta credentials. When the registry shows all services in green, Set up port-forwarding for the gateway as well.
+
+```bash
+$ kubectl port-forward svc/gateway -n demo 8080
+```
+
+Navigate to `http://localhost:8080`, sign in, and create some entities to verify everything is working fine.
+
+After looking around, stop minikube before the cloud deployment:
+
+```bash
+$ minikube stop
```

Comment:
I just tried this whole section with my MacBook Pro that's Intel and everything worked. I did have to wait for 10 minutes or so for everything to startup. You might recommend `minikube delete` here too since that helps make a clean slate for future tutorials.

Reply:
Added a the recommendation.